source/NVDAObjects/UIA/excel.py

```diff
@@ -0,0 +1,458 @@
+			states.add(controlTypes.STATE_HASPOPUP)
+		return states
+
+	def _get_cellCoordsText(self):
+		if self._hasSelection():
+			sc = self._getUIACacheablePropertyValue(
+				UIAHandler.UIA_SelectionItemSelectionContainerPropertyId
+			).QueryInterface(UIAHandler.IUIAutomationElement)
+
+			firstSelected = sc.GetCurrentPropertyValue(
+				UIAHandler.UIA_Selection2FirstSelectedItemPropertyId
+			).QueryInterface(UIAHandler.IUIAutomationElement)
+
+			firstAddress = firstSelected.GetCurrentPropertyValue(
+				UIAHandler.UIA_NamePropertyId
+			).replace('"', '')
+
+			firstValue = firstSelected.GetCurrentPropertyValue(
+				UIAHandler.UIA_ValueValuePropertyId
+			)
+
+			lastSelected = sc.GetCurrentPropertyValue(
+				UIAHandler.UIA_Selection2LastSelectedItemPropertyId
+			).QueryInterface(UIAHandler.IUIAutomationElement)
+
+			lastAddress = lastSelected.GetCurrentPropertyValue(
+				UIAHandler.UIA_NamePropertyId
+			).replace('"', '')
+
+			lastValue = lastSelected.GetCurrentPropertyValue(
+				UIAHandler.UIA_ValueValuePropertyId
+			)
+
+			cellCoordsTemplate = pgettext(
+				"excel-UIA",
+				# Translators: Excel, report range of cell coordinates
+				"{firstAddress} {firstValue} through {lastAddress} {lastValue}"
+			)
+			return cellCoordsTemplate.format(
+				firstAddress=firstAddress,
+				firstValue=firstValue,
+				lastAddress=lastAddress,
+				lastValue=lastValue
+			)
+		name = super().name
+		# Later builds of Excel 2016 quote the letter coordinate.
+		# We don't want the quotes.
+		name = name.replace('"', '')
+		return name
+
+	@script(
+		# Translators: the description  for a script for Excel
+		description=_("Reports the note or comment thread on the current cell"),
+		gesture="kb:NVDA+alt+c")
+	def script_reportComment(self, gesture):
+		commentsElement = self.UIAAnnotationObjects.get(UIAHandler.AnnotationType_Comment)
+		if commentsElement:
+			comment = commentsElement.GetCurrentPropertyValue(UIAHandler.UIA_FullDescriptionPropertyId)
+			author = commentsElement.GetCurrentPropertyValue(UIAHandler.UIA_AnnotationAuthorPropertyId)
+			numReplies = commentsElement.GetCurrentPropertyValue(self._UIAExcelCustomProps.commentReplyCount.id)
+			if numReplies == 0:
+				# Translators: a comment on a cell in Microsoft excel.
+				text = _("{comment}  by {author}").format(
+					comment=comment,
+					author=author
+				)
+			else:
+				# Translators: a comment on a cell in Microsoft excel.
+				text = _("{comment}  by {author} with {numReplies} replies").format(
+					comment=comment,
+					author=author,
+					numReplies=numReplies
+				)
+			ui.message(text)
+		else:
+			# Translators: A message in Excel when there is no note
+			ui.message(_("No note or comment thread on this cell"))
+
+
+class ExcelWorksheet(ExcelObject):
+	role = controlTypes.ROLE_TABLE
+
+	# The grid UIAElement dies each time the sheet is scrolled.
+	# therefore this grid would be announced in the focus ancestory each time which is bad.
```

Comment:
Typo, start with uppercase

```suggestion
	# Therefore this grid would be announced in the focus ancestory each time which is bad.
```